internal/mode/static/policies/clientsettings/generator_test.go

```diff
@@ -0,0 +1,173 @@
+	"github.com/nginxinc/nginx-gateway-fabric/internal/mode/static/policies/policiesfakes"
+)
+
+func TestGenerate(t *testing.T) {
+	maxSize := helpers.GetPointer[ngfAPI.Size]("10m")
+	bodyTimeout := helpers.GetPointer[ngfAPI.Duration]("600ms")
+	keepaliveRequests := helpers.GetPointer[int32](900)
+	keepaliveTime := helpers.GetPointer[ngfAPI.Duration]("50s")
+	keepaliveServerTimeout := helpers.GetPointer[ngfAPI.Duration]("30s")
+	keepaliveHeaderTimeout := helpers.GetPointer[ngfAPI.Duration]("60s")
+
+	tests := []struct {
+		name       string
+		policy     policies.Policy
+		expStrings []string
+	}{
+		{
+			name: "body max size populated",
+			policy: &ngfAPI.ClientSettingsPolicy{
+				Spec: ngfAPI.ClientSettingsPolicySpec{
+					Body: &ngfAPI.ClientBody{
+						MaxSize: maxSize,
+					},
+				},
+			},
+			expStrings: []string{
+				"client_max_body_size 10m;",
+			},
+		},
+		{
+			name: "body timeout populated",
+			policy: &ngfAPI.ClientSettingsPolicy{
+				Spec: ngfAPI.ClientSettingsPolicySpec{
+					Body: &ngfAPI.ClientBody{
+						Timeout: bodyTimeout,
+					},
+				},
+			},
+			expStrings: []string{
+				"client_body_timeout 600ms",
+			},
+		},
+		{
+			name: "keepalive requests populated",
+			policy: &ngfAPI.ClientSettingsPolicy{
+				Spec: ngfAPI.ClientSettingsPolicySpec{
+					KeepAlive: &ngfAPI.ClientKeepAlive{
+						Requests: keepaliveRequests,
+					},
+				},
+			},
+			expStrings: []string{
+				"keepalive_requests 900;",
+			},
+		},
+		{
+			name: "keepalive time populated",
+			policy: &ngfAPI.ClientSettingsPolicy{
+				Spec: ngfAPI.ClientSettingsPolicySpec{
+					KeepAlive: &ngfAPI.ClientKeepAlive{
+						Time: keepaliveTime,
+					},
+				},
+			},
+			expStrings: []string{
+				"keepalive_time 50s;",
+			},
+		},
+		{
+			name: "keepalive timeout server populated",
+			policy: &ngfAPI.ClientSettingsPolicy{
+				Spec: ngfAPI.ClientSettingsPolicySpec{
+					KeepAlive: &ngfAPI.ClientKeepAlive{
+						Timeout: &ngfAPI.ClientKeepAliveTimeout{
+							Server: keepaliveServerTimeout,
+						},
+					},
+				},
+			},
+			expStrings: []string{
+				"keepalive_timeout 30s;",
+			},
+		},
+		{
+			name: "keepalive timeout server and header populated",
+			policy: &ngfAPI.ClientSettingsPolicy{
+				Spec: ngfAPI.ClientSettingsPolicySpec{
+					KeepAlive: &ngfAPI.ClientKeepAlive{
+						Timeout: &ngfAPI.ClientKeepAliveTimeout{
+							Server: keepaliveServerTimeout,
+							Header: keepaliveHeaderTimeout,
+						},
+					},
+				},
+			},
+			expStrings: []string{
+				"keepalive_timeout 30s 60s;",
+			},
+		},
+		{
+			name: "keepalive timeout header populated",
+			policy: &ngfAPI.ClientSettingsPolicy{
+				Spec: ngfAPI.ClientSettingsPolicySpec{
+					KeepAlive: &ngfAPI.ClientKeepAlive{
+						Timeout: &ngfAPI.ClientKeepAliveTimeout{
+							Header: keepaliveHeaderTimeout,
+						},
+					},
+				},
+			},
+			expStrings: []string{}, // header timeout is ignored if server timeout is not populated
```

Comment:
It is handled by CEL validation. I just am also handling it in the template.

Reply:
I worry that we now spread validation logic to more places like template, and it's gonna be harder to maintain and extend such code. What if we just not handle that logic in the template and accept consequences of CRD validation failure? Will it just be invalid NGINX conf?

Reply:
This is a weird case. If server timeout isn't provided we can't just omit it:

`keepalive_timeout   30s` 

because that will just end up setting the server timeout to 30s. 

We could choose to either:
1. Set server timeout to something invalid like `""` which will cause nginx to fail
2. Set server timeout to the default `75s`.

Option (2) seems like the obvious choice, but it does go against our plan of omitting default nginx values so we don't have to update them when they change.

Reply:
huh. `keepalive_timeout   30s` won't cause a reload failures. But other options seems like no matter way we choose, the config generation or template will have some logic that handles the case of failed validation.

I wonder if it will be easier just to add that validation in the validation layer for that case (with the corresponding explanation why), so that it is in validation layer, just for that reason. and remove edge case handling from template.

Reply:
Seems reasonable to me. @sjberman what did you think?

Reply:
Sounds good to me

Reply:
Added back the validation for this